netkan/netkan/metadata.py

```diff
@@ -0,0 +1,48 @@
+import json
+import re
+from pathlib import Path
+from hashlib import md5
+
+
+class Netkan:
+
+    kref_pattern = re.compile('^#/ckan/([^/]+)/(.+)$')
+
+    def __init__(self, filename):
+        self.filename = Path(filename)
+        self.contents = self.filename.read_text()
+        self._raw = json.loads(self.contents)
+
+    def __getattr__(self, name):
```

Comment:
This is going to make for difficult maintenance. If a developer needs to check what `on_spacedock` does to investigate a future bug, it's going to take a lot of reading and parsing and guessing.

Can we stick to explicitly naming functions with `def` the normal way, so they can be found?

Reply:
This is a relatively common pattern in Python and means if we add a new kref type it's automatically supported by the class. The tests cover ensuring the variations work, though a comment or two probably wouldn't go astray.

I'm not 100% happy with the `__getattr__` block, but had to leave for dance class!

Reply:
I suppose the more serious problem is that it obscures the class's interface. If I have a reference to a `Netkan` object, the most important and frequent question I will ask is, "What can I do with this object"? With `def`'d functions and public variables assigned in the constructor, the answers are clear. With `__getattr__`, they could hardly be more murky. Imagine having to discover the existence of `on_curse` if you didn't already know exactly how this `__getattr__` implementation worked.

For "add a new kref type it's automatically supported by the class", I recommend making it an explicit parameter:

```python
    def is_on(kind):
        return kind == self.kref.kind # (or whatever implementation makes sense)
```

Reply:
This as often a frustration with an undocumented class, but the reason behind this will become more apparent with the download counter. In big apps the maintenance burden is usually where the explicit patterns start becoming a burden as an interface change is more than a one liner.

I'll add some documentation when I get to work. I think the logic behind it will make more sense as I lean harder on the class.